src/Compilers/CSharp/Portable/Binder/Binder_Constraints.cs

```diff
@@ -214,12 +250,26 @@ private TypeParameterConstraintClause BindTypeParameterConstraints(TypeParameter
                 }
             }
 
-            if (!hasTypeLikeConstraint && !IsNullableEnabled(typeParameterSyntax.Identifier))
+            if (!isForOverride && !hasTypeLikeConstraint && !IsNullableEnabled(typeParameterSyntax.Identifier))
             {
                 constraints |= TypeParameterConstraintKind.ObliviousNullabilityIfReferenceType;
             }
 
-            return TypeParameterConstraintClause.Create(constraints, constraintTypes.ToImmutableAndFree(), syntaxBuilder.ToImmutableAndFree());
+            Debug.Assert(!isForOverride ||
+                         (constraints & (TypeParameterConstraintKind.ReferenceType | TypeParameterConstraintKind.ValueType)) != (TypeParameterConstraintKind.ReferenceType | TypeParameterConstraintKind.ValueType));
+
+            return TypeParameterConstraintClause.Create(constraints,
+                                                        constraintTypes?.ToImmutableAndFree() ?? ImmutableArray<TypeWithAnnotations>.Empty,
+                                                        syntaxBuilder?.ToImmutableAndFree() ?? default);
```

Comment:
>default [](start = 95, length = 7)

Feels like this should be an empty array as well

Reply:
> Feels like this should be an empty array as well

No,  ```TypeParameterConstraintClause.Create``` expects it to be ```default``` when the previous parameter is empty.

---
In reply to: [276458251](https://github.com/dotnet/roslyn/pull/35048#discussion_r276458251) [](ancestors = 276458251)